From 4d0180d2d0c39e7c111796255b273ef5be11634c Mon Sep 17 00:00:00 2001
From: Michael D Labriola <mdlabriola@yahoo.com>
Date: Sat, 14 Aug 2010 09:38:47 -0400
Subject: [PATCH 1/3] use linker from /bootstrap/lib

This commit is a modified version of the gcc-spec patch from the CLFS book.
The only difference is that /tools was replaced with /bootstrap
---
 gcc/config/alpha/gnu.h        |    2 +-
 gcc/config/alpha/linux-elf.h  |    4 ++--
 gcc/config/arm/linux-eabi.h   |    2 +-
 gcc/config/arm/linux-elf.h    |    2 +-
 gcc/config/bfin/linux.h       |    2 +-
 gcc/config/cris/linux.h       |    2 +-
 gcc/config/freebsd-spec.h     |    2 +-
 gcc/config/frv/linux.h        |    2 +-
 gcc/config/i386/gnu.h         |    2 +-
 gcc/config/i386/linux.h       |    2 +-
 gcc/config/i386/linux64.h     |    4 ++--
 gcc/config/ia64/linux.h       |    2 +-
 gcc/config/kfreebsd-gnu.h     |    2 +-
 gcc/config/knetbsd-gnu.h      |    2 +-
 gcc/config/kopensolaris-gnu.h |    2 +-
 gcc/config/linux.h            |    6 +++---
 gcc/config/lm32/uclinux-elf.h |    2 +-
 gcc/config/m32r/linux.h       |    2 +-
 gcc/config/m68k/linux.h       |    2 +-
 gcc/config/mips/linux.h       |    2 +-
 gcc/config/mips/linux64.h     |    8 ++++----
 gcc/config/mn10300/linux.h    |    2 +-
 gcc/config/pa/pa-linux.h      |    2 +-
 gcc/config/rs6000/linux64.h   |    8 ++++----
 gcc/config/rs6000/sysv4.h     |    6 +++---
 gcc/config/s390/linux.h       |    4 ++--
 gcc/config/sh/linux.h         |    2 +-
 gcc/config/sparc/linux.h      |    4 ++--
 gcc/config/sparc/linux64.h    |   10 +++++-----
 gcc/config/vax/linux.h        |    2 +-
 gcc/config/xtensa/linux.h     |    2 +-
 31 files changed, 49 insertions(+), 49 deletions(-)

diff --git a/gcc/config/alpha/gnu.h b/gcc/config/alpha/gnu.h
index f98c3e7..5a406b7 100644
--- a/gcc/config/alpha/gnu.h
+++ b/gcc/config/alpha/gnu.h
@@ -34,7 +34,7 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
     } while (0)
 
 #undef ELF_DYNAMIC_LINKER
-#define ELF_DYNAMIC_LINKER	"/lib/ld.so"
+#define ELF_DYNAMIC_LINKER	"/bootstrap/lib/ld.so"
 
 #undef	STARTFILE_SPEC
 #define STARTFILE_SPEC \
diff --git a/gcc/config/alpha/linux-elf.h b/gcc/config/alpha/linux-elf.h
index c48fc93..c3c79d0 100644
--- a/gcc/config/alpha/linux-elf.h
+++ b/gcc/config/alpha/linux-elf.h
@@ -27,8 +27,8 @@ along with GCC; see the file COPYING3.  If not see
 #define EXTRA_SPECS \
 { "elf_dynamic_linker", ELF_DYNAMIC_LINKER },
 
-#define GLIBC_DYNAMIC_LINKER	"/lib/ld-linux.so.2"
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER	"/bootstrap/lib/ld-linux.so.2"
+#define UCLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld-uClibc.so.0"
 #if UCLIBC_DEFAULT
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:%{muclibc:%e-mglibc and -muclibc used together}" G ";:" U "}"
 #else
diff --git a/gcc/config/arm/linux-eabi.h b/gcc/config/arm/linux-eabi.h
index fce1ed1..5d97f8f 100644
--- a/gcc/config/arm/linux-eabi.h
+++ b/gcc/config/arm/linux-eabi.h
@@ -61,7 +61,7 @@
 /* Use ld-linux.so.3 so that it will be possible to run "classic"
    GNU/Linux binaries on an EABI system.  */
 #undef  GLIBC_DYNAMIC_LINKER
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.3"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld-linux.so.3"
 
 /* At this point, bpabi.h will have clobbered LINK_SPEC.  We want to
    use the GNU/Linux version, not the generic BPABI version.  */
diff --git a/gcc/config/arm/linux-elf.h b/gcc/config/arm/linux-elf.h
index 9fdca41..2f6057b 100644
--- a/gcc/config/arm/linux-elf.h
+++ b/gcc/config/arm/linux-elf.h
@@ -62,7 +62,7 @@
 
 #define LIBGCC_SPEC "%{msoft-float:-lfloat} %{mfloat-abi=soft*:-lfloat} -lgcc"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld-linux.so.2"
 
 #define LINUX_TARGET_LINK_SPEC  "%{h*} %{version:-v} \
    %{b} \
diff --git a/gcc/config/bfin/linux.h b/gcc/config/bfin/linux.h
index 5c716cd..3c91229 100644
--- a/gcc/config/bfin/linux.h
+++ b/gcc/config/bfin/linux.h
@@ -45,7 +45,7 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
   %{shared:-G -Bdynamic} \
   %{!shared: %{!static: \
    %{rdynamic:-export-dynamic} \
-   %{!dynamic-linker:-dynamic-linker /lib/ld-uClibc.so.0}} \
+   %{!dynamic-linker:-dynamic-linker /bootstrap/lib/ld-uClibc.so.0}} \
    %{static}} -init __init -fini __fini"
 
 #define MD_UNWIND_SUPPORT "config/bfin/linux-unwind.h"
diff --git a/gcc/config/cris/linux.h b/gcc/config/cris/linux.h
index 38f0a7e..38f48d6 100644
--- a/gcc/config/cris/linux.h
+++ b/gcc/config/cris/linux.h
@@ -103,7 +103,7 @@ along with GCC; see the file COPYING3.  If not see
 #undef CRIS_DEFAULT_CPU_VERSION
 #define CRIS_DEFAULT_CPU_VERSION CRIS_CPU_NG
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld.so.1"
 
 /* We need an -rpath-link to ld.so.1, and presumably to each directory
    specified with -B.  */
diff --git a/gcc/config/freebsd-spec.h b/gcc/config/freebsd-spec.h
index 84d8127..80beb84 100644
--- a/gcc/config/freebsd-spec.h
+++ b/gcc/config/freebsd-spec.h
@@ -146,7 +146,7 @@ is built with the --enable-threads configure-time option.}		\
 #if FBSD_MAJOR < 6
 #define FBSD_DYNAMIC_LINKER "/usr/libexec/ld-elf.so.1"
 #else
-#define FBSD_DYNAMIC_LINKER "/libexec/ld-elf.so.1"
+#define FBSD_DYNAMIC_LINKER "/bootstrap/libexec/ld-elf.so.1"
 #endif
 
 #if defined(HAVE_LD_EH_FRAME_HDR)
diff --git a/gcc/config/frv/linux.h b/gcc/config/frv/linux.h
index 71cd094..ac98304 100644
--- a/gcc/config/frv/linux.h
+++ b/gcc/config/frv/linux.h
@@ -35,7 +35,7 @@
 #define ENDFILE_SPEC \
   "%{shared|pie:crtendS.o%s;:crtend.o%s} crtn.o%s"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC "\
diff --git a/gcc/config/i386/gnu.h b/gcc/config/i386/gnu.h
index 077933b..a9ab90c 100644
--- a/gcc/config/i386/gnu.h
+++ b/gcc/config/i386/gnu.h
@@ -21,7 +21,7 @@ along with GCC.  If not, see <http://www.gnu.org/licenses/>.
 */
 
 #undef GLIBC_DYNAMIC_LINKER
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld.so"
 
 #undef TARGET_VERSION
 #define TARGET_VERSION fprintf (stderr, " (i386 GNU)");
diff --git a/gcc/config/i386/linux.h b/gcc/config/i386/linux.h
index 2a31880..cbf5030 100644
--- a/gcc/config/i386/linux.h
+++ b/gcc/config/i386/linux.h
@@ -100,7 +100,7 @@ along with GCC; see the file COPYING3.  If not see
 
 /* These macros may be overridden in k*bsd-gnu.h and i386/k*bsd-gnu.h. */
 #define LINK_EMULATION "elf_i386"
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld-linux.so.2"
 
 #undef  ASM_SPEC
 #define ASM_SPEC \
diff --git a/gcc/config/i386/linux64.h b/gcc/config/i386/linux64.h
index 867de59..8f26e65 100644
--- a/gcc/config/i386/linux64.h
+++ b/gcc/config/i386/linux64.h
@@ -58,8 +58,8 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
    When the -shared link option is used a final link is not being
    done.  */
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux-x86-64.so.2"
+#define GLIBC_DYNAMIC_LINKER32 "/bootstrap/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/bootstrap/lib64/ld-linux-x86-64.so.2"
 
 #if TARGET_64BIT_DEFAULT
 #define SPEC_32 "m32"
diff --git a/gcc/config/ia64/linux.h b/gcc/config/ia64/linux.h
index 587a4f4..1f4bb28 100644
--- a/gcc/config/ia64/linux.h
+++ b/gcc/config/ia64/linux.h
@@ -61,7 +61,7 @@ do {						\
 /* Define this for shared library support because it isn't in the main
    linux.h file.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux-ia64.so.2"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld-linux-ia64.so.2"
 
 #undef LINK_SPEC
 #define LINK_SPEC "\
diff --git a/gcc/config/kfreebsd-gnu.h b/gcc/config/kfreebsd-gnu.h
index 592bae3..aa5e7b8 100644
--- a/gcc/config/kfreebsd-gnu.h
+++ b/gcc/config/kfreebsd-gnu.h
@@ -33,5 +33,5 @@ along with GCC; see the file COPYING3.  If not see
 
 #ifdef GLIBC_DYNAMIC_LINKER
 #undef GLIBC_DYNAMIC_LINKER
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld.so.1"
 #endif
diff --git a/gcc/config/knetbsd-gnu.h b/gcc/config/knetbsd-gnu.h
index 4cc8eb7..d23ee8f 100644
--- a/gcc/config/knetbsd-gnu.h
+++ b/gcc/config/knetbsd-gnu.h
@@ -34,5 +34,5 @@ along with GCC; see the file COPYING3.  If not see
 
 #ifdef GLIBC_DYNAMIC_LINKER
 #undef GLIBC_DYNAMIC_LINKER
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld.so.1"
 #endif
diff --git a/gcc/config/kopensolaris-gnu.h b/gcc/config/kopensolaris-gnu.h
index e044faf..4bc6a1a 100644
--- a/gcc/config/kopensolaris-gnu.h
+++ b/gcc/config/kopensolaris-gnu.h
@@ -33,5 +33,5 @@ along with GCC; see the file COPYING3.  If not see
 
 #ifdef GLIBC_DYNAMIC_LINKER
 #undef GLIBC_DYNAMIC_LINKER
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld.so.1"
 #endif
diff --git a/gcc/config/linux.h b/gcc/config/linux.h
index e567e9a..7908af7 100644
--- a/gcc/config/linux.h
+++ b/gcc/config/linux.h
@@ -117,9 +117,9 @@ see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
    GLIBC_DYNAMIC_LINKER must be defined for each target using them, or
    GLIBC_DYNAMIC_LINKER32 and GLIBC_DYNAMIC_LINKER64 for targets
    supporting both 32-bit and 64-bit compilation.  */
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER32 "/bootstrap/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER64 "/bootstrap/lib/ld64-uClibc.so.0"
 #define LINUX_DYNAMIC_LINKER \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKER, UCLIBC_DYNAMIC_LINKER)
 #define LINUX_DYNAMIC_LINKER32 \
diff --git a/gcc/config/lm32/uclinux-elf.h b/gcc/config/lm32/uclinux-elf.h
index ce9719a..ad31d82 100644
--- a/gcc/config/lm32/uclinux-elf.h
+++ b/gcc/config/lm32/uclinux-elf.h
@@ -73,7 +73,7 @@
    %{shared:-shared} \
    %{symbolic:-Bsymbolic} \
    %{rdynamic:-export-dynamic} \
-   %{!dynamic-linker:-dynamic-linker /lib/ld-linux.so.2}"
+   %{!dynamic-linker:-dynamic-linker /bootstrap/lib/ld-linux.so.2}"
 
 #define TARGET_OS_CPP_BUILTINS() LINUX_TARGET_OS_CPP_BUILTINS()
 
diff --git a/gcc/config/m32r/linux.h b/gcc/config/m32r/linux.h
index 0515e5b..0264661 100644
--- a/gcc/config/m32r/linux.h
+++ b/gcc/config/m32r/linux.h
@@ -58,7 +58,7 @@
 
 /* If ELF is the default format, we should not use /lib/elf.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld-linux.so.2"
 
 #undef	LINK_SPEC
 #if TARGET_LITTLE_ENDIAN
diff --git a/gcc/config/m68k/linux.h b/gcc/config/m68k/linux.h
index 1d73ee0..2dadef7 100644
--- a/gcc/config/m68k/linux.h
+++ b/gcc/config/m68k/linux.h
@@ -84,7 +84,7 @@ along with GCC; see the file COPYING3.  If not see
 
 /* If ELF is the default format, we should not use /lib/elf.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC "-m m68kelf %{shared} \
diff --git a/gcc/config/mips/linux.h b/gcc/config/mips/linux.h
index 0512ef7..229ab76 100644
--- a/gcc/config/mips/linux.h
+++ b/gcc/config/mips/linux.h
@@ -64,7 +64,7 @@ along with GCC; see the file COPYING3.  If not see
 #undef MIPS_DEFAULT_GVALUE
 #define MIPS_DEFAULT_GVALUE 0
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld.so.1"
 
 /* Borrowed from sparc/linux.h */
 #undef LINK_SPEC
diff --git a/gcc/config/mips/linux64.h b/gcc/config/mips/linux64.h
index 2f24dfa..e20017b 100644
--- a/gcc/config/mips/linux64.h
+++ b/gcc/config/mips/linux64.h
@@ -35,10 +35,10 @@ along with GCC; see the file COPYING3.  If not see
 %{!shared: \
   %{profile:-lc_p} %{!profile:-lc}}"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld.so.1"
-#define GLIBC_DYNAMIC_LINKERN32 "/lib32/ld.so.1"
-#define UCLIBC_DYNAMIC_LINKERN32 "/lib32/ld-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER32 "/bootstrap/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/bootstrap/lib64/ld.so.1"
+#define GLIBC_DYNAMIC_LINKERN32 "/bootstrap/lib32/ld.so.1"
+#define UCLIBC_DYNAMIC_LINKERN32 "/bootstrap/lib32/ld-uClibc.so.0"
 #define LINUX_DYNAMIC_LINKERN32 \
   CHOOSE_DYNAMIC_LINKER (GLIBC_DYNAMIC_LINKERN32, UCLIBC_DYNAMIC_LINKERN32)
 
diff --git a/gcc/config/mn10300/linux.h b/gcc/config/mn10300/linux.h
index ac9f8c5..de2f312 100644
--- a/gcc/config/mn10300/linux.h
+++ b/gcc/config/mn10300/linux.h
@@ -32,7 +32,7 @@
 #undef  ASM_SPEC
 #define ASM_SPEC "%{Wa,*:%*}"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld.so.1"
 
 #undef  LINK_SPEC
 #define LINK_SPEC "%{mrelax:--relax} %{shared:-shared} \
diff --git a/gcc/config/pa/pa-linux.h b/gcc/config/pa/pa-linux.h
index cd46790..1439882 100644
--- a/gcc/config/pa/pa-linux.h
+++ b/gcc/config/pa/pa-linux.h
@@ -38,7 +38,7 @@ along with GCC; see the file COPYING3.  If not see
 /* Define this for shared library support because it isn't in the main
    linux.h file.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC "\
diff --git a/gcc/config/rs6000/linux64.h b/gcc/config/rs6000/linux64.h
index da7bca3..eeaecc0 100644
--- a/gcc/config/rs6000/linux64.h
+++ b/gcc/config/rs6000/linux64.h
@@ -348,10 +348,10 @@ extern int dot_symbols;
 #undef	LINK_OS_DEFAULT_SPEC
 #define LINK_OS_DEFAULT_SPEC "%(link_os_linux)"
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld64.so.1"
-#define UCLIBC_DYNAMIC_LINKER32 "/lib/ld-uClibc.so.0"
-#define UCLIBC_DYNAMIC_LINKER64 "/lib/ld64-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER32 "/bootstrap/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/bootstrap/lib64/ld64.so.1"
+#define UCLIBC_DYNAMIC_LINKER32 "/bootstrap/lib/ld-uClibc.so.0"
+#define UCLIBC_DYNAMIC_LINKER64 "/bootstrap/lib/ld64-uClibc.so.0"
 #if UCLIBC_DEFAULT
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:%{muclibc:%e-mglibc and -muclibc used together}" G ";:" U "}"
 #else
diff --git a/gcc/config/rs6000/sysv4.h b/gcc/config/rs6000/sysv4.h
index 8d19ad7..8226bd9 100644
--- a/gcc/config/rs6000/sysv4.h
+++ b/gcc/config/rs6000/sysv4.h
@@ -901,8 +901,8 @@ SVR4_ASM_SPEC \
 
 #define LINK_START_LINUX_SPEC ""
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
-#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld.so.1"
+#define UCLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld-uClibc.so.0"
 #if UCLIBC_DEFAULT
 #define CHOOSE_DYNAMIC_LINKER(G, U) "%{mglibc:%{muclibc:%e-mglibc and -muclibc used together}" G ";:" U "}"
 #else
@@ -944,7 +944,7 @@ SVR4_ASM_SPEC \
 
 #define LINK_OS_GNU_SPEC "-m elf32ppclinux %{!shared: %{!static: \
   %{rdynamic:-export-dynamic} \
-  %{!dynamic-linker:-dynamic-linker /lib/ld.so.1}}}"
+  %{!dynamic-linker:-dynamic-linker /bootstrap/lib/ld.so.1}}}"
 
 #define CPP_OS_GNU_SPEC "-D__unix__ -D__gnu_hurd__ -D__GNU__	\
 %{!undef:					                \
diff --git a/gcc/config/s390/linux.h b/gcc/config/s390/linux.h
index ddbcfa9..aeaa00d 100644
--- a/gcc/config/s390/linux.h
+++ b/gcc/config/s390/linux.h
@@ -72,8 +72,8 @@ along with GCC; see the file COPYING3.  If not see
 #define MULTILIB_DEFAULTS { "m31" }
 #endif
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld.so.1"
-#define GLIBC_DYNAMIC_LINKER64 "/lib/ld64.so.1"
+#define GLIBC_DYNAMIC_LINKER32 "/bootstrap/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER64 "/bootstrap/lib/ld64.so.1"
 
 #undef  LINK_SPEC
 #define LINK_SPEC \
diff --git a/gcc/config/sh/linux.h b/gcc/config/sh/linux.h
index d513af8..bd377d9 100644
--- a/gcc/config/sh/linux.h
+++ b/gcc/config/sh/linux.h
@@ -47,7 +47,7 @@ along with GCC; see the file COPYING3.  If not see
 
 #define TARGET_ASM_FILE_END file_end_indicate_exec_stack
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld-linux.so.2"
 
 #undef SUBTARGET_LINK_EMUL_SUFFIX
 #define SUBTARGET_LINK_EMUL_SUFFIX "_linux"
diff --git a/gcc/config/sparc/linux.h b/gcc/config/sparc/linux.h
index b12c881..9fe2d1f 100644
--- a/gcc/config/sparc/linux.h
+++ b/gcc/config/sparc/linux.h
@@ -83,10 +83,10 @@ along with GCC; see the file COPYING3.  If not see
 
 /* If ELF is the default format, we should not use /lib/elf.  */
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld-linux.so.2"
 
 #undef  LINK_SPEC
-#define LINK_SPEC "-m elf32_sparc -Y P,/usr/lib %{shared:-shared} \
+#define LINK_SPEC "-m elf32_sparc -Y P,/bootstrap/lib %{shared:-shared} \
   %{!mno-relax:%{!r:-relax}} \
   %{!shared: \
     %{!ibcs: \
diff --git a/gcc/config/sparc/linux64.h b/gcc/config/sparc/linux64.h
index c2e78d7..274b354 100644
--- a/gcc/config/sparc/linux64.h
+++ b/gcc/config/sparc/linux64.h
@@ -109,8 +109,8 @@ along with GCC; see the file COPYING3.  If not see
 
 /* If ELF is the default format, we should not use /lib/elf.  */
 
-#define GLIBC_DYNAMIC_LINKER32 "/lib/ld-linux.so.2"
-#define GLIBC_DYNAMIC_LINKER64 "/lib64/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER32 "/bootstrap/lib/ld-linux.so.2"
+#define GLIBC_DYNAMIC_LINKER64 "/bootstrap/lib64/ld-linux.so.2"
 
 #ifdef SPARC_BI_ARCH
 
@@ -121,7 +121,7 @@ along with GCC; see the file COPYING3.  If not see
   { "link_arch_default", LINK_ARCH_DEFAULT_SPEC },	  \
   { "link_arch",	 LINK_ARCH_SPEC },
 
-#define LINK_ARCH32_SPEC "-m elf32_sparc -Y P,%R/usr/lib %{shared:-shared} \
+#define LINK_ARCH32_SPEC "-m elf32_sparc -Y P,%R/bootstrap/lib %{shared:-shared} \
   %{!shared: \
     %{!ibcs: \
       %{!static: \
@@ -130,7 +130,7 @@ along with GCC; see the file COPYING3.  If not see
         %{static:-static}}} \
 "
 
-#define LINK_ARCH64_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_ARCH64_SPEC "-m elf64_sparc -Y P,%R/bootstrap/lib64 %{shared:-shared} \
   %{!shared: \
     %{!ibcs: \
       %{!static: \
@@ -211,7 +211,7 @@ along with GCC; see the file COPYING3.  If not see
 #else /* !SPARC_BI_ARCH */
 
 #undef LINK_SPEC
-#define LINK_SPEC "-m elf64_sparc -Y P,%R/usr/lib64 %{shared:-shared} \
+#define LINK_SPEC "-m elf64_sparc -Y P,%R/bootstrap/lib64 %{shared:-shared} \
   %{!shared: \
     %{!ibcs: \
       %{!static: \
diff --git a/gcc/config/vax/linux.h b/gcc/config/vax/linux.h
index dccbe9c..c301898 100644
--- a/gcc/config/vax/linux.h
+++ b/gcc/config/vax/linux.h
@@ -44,5 +44,5 @@ along with GCC; see the file COPYING3.  If not see
   %{!shared: \
     %{!static: \
       %{rdynamic:-export-dynamic} \
-      %{!dynamic-linker:-dynamic-linker /lib/ld.so.1}} \
+      %{!dynamic-linker:-dynamic-linker /bootstrap/lib/ld.so.1}} \
     %{static:-static}}"
diff --git a/gcc/config/xtensa/linux.h b/gcc/config/xtensa/linux.h
index 69ab626..7b1efc3 100644
--- a/gcc/config/xtensa/linux.h
+++ b/gcc/config/xtensa/linux.h
@@ -43,7 +43,7 @@ along with GCC; see the file COPYING3.  If not see
   %{mlongcalls:--longcalls} \
   %{mno-longcalls:--no-longcalls}"
 
-#define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
+#define GLIBC_DYNAMIC_LINKER "/bootstrap/lib/ld.so.1"
 
 #undef LINK_SPEC
 #define LINK_SPEC \
-- 
1.7.1.1

